lib/workers/repository/extract/index.ts

```diff
@@ -30,13 +32,17 @@ export async function extractAllDependencies(
   };
 
   for (const manager of managerList) {
-    const managerConfig = getManagerConfig(config, manager);
-    managerConfig.manager = manager;
-    if (manager === 'regex') {
-      for (const regexManager of config.regexManagers ?? []) {
-        tryConfig(mergeChildConfig(managerConfig, regexManager));
+    if (manager === 'custom') {
+      for (const regexManager of config.regexManagers ??
+        // istanbul ignore next
+        []) {
+        const customManagerConfig = getManagerConfig(config, 'regex'); // TODO: replace 'regex' with regexManager.customType
+        customManagerConfig.manager = 'regex'; // TODO: replace 'regex' with regexManager.customType
+        tryConfig(mergeChildConfig(customManagerConfig, regexManager));
       }
-    } else {
+    } else if (!customManagerList.includes(manager)) {
+      const managerConfig = getManagerConfig(config, manager);
+      managerConfig.manager = manager;
       tryConfig(managerConfig);
     }
```

Comment:
There is an undiscussed change here, because there might be other ways to do it too:

In the extract fn during this loop:
The actual custom manager is extracted here and then passed down. So in all fns after this like `extractPackageFile`, `lookup`, `branchify` and `update` the manager will be `regex` and `jsonata` and not `custom`. Is this okay?

I thought passing down `customType` and using it inplace of manager these functions might only increase complexity without adding much benefits.

Reply:
wouldn't this change package rule apply logic? 🤔 eg custom no longer applies as match manager after extract.

Reply:
It should be fine because of the way custom managers are handled in matchManagers.
If manager is say `regex` we first check if the matchManagers array has `custom` too, instead checking for `regex` only.

Reply:
I'm not sure that this is the best idea, although maybe it's ok. @secustor wdyt?

Reply:
So if I understand this correctly the user can match `custom` and sub variants e.g. `regex` using `matchManager`.
 
This should work on first glance, tough I would rather add all sub variants in the style of `custom.regex` to the list.    
That makes it easier understandable that custom and regex are connected and not totally different managers, also this should remove the need for special handling in extract code too.

Further from an UX perspective that would make it more consistent with `customDatasources`.

Reply:
Yes I like the custom.regex

Reply:
```ts
matchManagers = ['custom.regex', 'custom.jsonata']  // allowed
matchManagers = ['custom', 'regex', 'jsonata'] // not allowed
```

And do we want the same for `enabledManagers`?

Reply:
I like the `custom.*` variant 👍

Reply:
Should this be the way things are handled internally as well?
ie. in the packageFiles with  updates or managerConfig we return manager as `custom.regex` instead of `regex`

Reply:
yes, I think it's acceptable.

Reply:
Sounds fine